pkg/destroy/powervs/cloudconnection.go

```diff
@@ -34,29 +54,73 @@ func (o *ClusterUninstaller) listCloudConnections() (cloudResources, error) {
 		log.Fatalf("Failed to list cloud connections: %v", err)
 	}
 
-	var foundOne = false
-
 	result := []cloudResource{}
 	for _, cloudConnection = range cloudConnections.CloudConnections {
-		if strings.Contains(*cloudConnection.Name, o.InfraID) {
-			o.Logger.Debugf("listCloudConnections: FOUND: %s (%s)", *cloudConnection.Name, *cloudConnection.CloudConnectionID)
-			foundOne = true
+		if !strings.Contains(*cloudConnection.Name, o.InfraID) {
+			// Skip this one!
+			continue
+		}
+
+		foundOne = true
+
+		o.Logger.Debugf("listCloudConnections: FOUND: %s (%s)", *cloudConnection.Name, *cloudConnection.CloudConnectionID)
+
+		cloudConnectionID = *cloudConnection.CloudConnectionID
+
+		vpcStillExists = true
+
+		for !o.timeout(ctx) {
+			if !vpcStillExists {
+				break
+			}
 
-			jobReference, err = o.cloudConnectionClient.Delete(*cloudConnection.CloudConnectionID)
+			select {
+			case <-o.Context.Done():
+				o.Logger.Debugf("listCloudConnections: case <-o.Context.Done()")
+				return nil, o.Context.Err() // we're cancelled, abort
+			default:
+			}
+
+			cloudConnection, err = o.cloudConnectionClient.Get(cloudConnectionID)
 			if err != nil {
-				errors.Errorf("Failed to delete cloud connection (%s): %v", *cloudConnection.CloudConnectionID, err)
+				log.Fatalf("Failed to get cloud connection %s: %v", cloudConnectionID, err)
 			}
 
-			o.Logger.Debugf("listCloudConnections: jobReference.ID = %s\n", *jobReference.ID)
+			EndpointVpc = cloudConnection.Vpc
+			o.Logger.Debugf("listCloudConnections: EndpointVpc = %+v", EndpointVpc)
+
+			foundVpc = false
+			for _, Vpc = range EndpointVpc.Vpcs {
+				if Vpc != nil {
+					foundVpc = true
+				}
+				o.Logger.Debugf("listCloudConnections: Vpc = %+v", Vpc)
+			}
+			o.Logger.Debugf("listCloudConnections: foundVpc = %v", foundVpc)
+			if foundVpc {
+				o.Logger.Debugf("listCloudConnections: This CC still has VPCs attached, waiting...")
 
-			result = append(result, cloudResource{
-				key:      *jobReference.ID,
-				name:     *jobReference.ID,
-				status:   "",
-				typeName: jobTypeName,
-				id:       *jobReference.ID,
-			})
+				time.Sleep(15 * time.Second)
+			} else {
+				vpcStillExists = false
+			}
+		}
+
+		// Finally delete the CloudConnection!
+		jobReference, err = o.cloudConnectionClient.Delete(*cloudConnection.CloudConnectionID)
```

Comment:
Nit:
```suggestion
		jobReference, err := o.cloudConnectionClient.Delete(*cloudConnection.CloudConnectionID)
```
Since this variable is only used here. No need to declare it a function level.